tests/test_proxy_functional.py

```diff
@@ -3,13 +3,52 @@
 import pathlib
 from unittest import mock
 
+import proxy
 import pytest
+import trustme
 from yarl import URL
 
 import aiohttp
 from aiohttp import web
 
 
+@pytest.fixture
+def trustme_certs(tmp_path):
+    ca = trustme.CA()
+    ca.issue_cert("localhost")
+    ca_cert_path = tmp_path / "ca.pem"
+    ca.cert_pem.write_to_path(ca_cert_path)
+    ca_key_path = tmp_path / "server.key"
+    ca.private_key_pem.write_to_path(ca_key_path)
+    return (str(ca_cert_path), str(ca_key_path))
+
+
+@pytest.fixture
+def secure_proxy_url(trustme_certs):
+    port = proxy.common.utils.get_available_port()
+    input_args = [
+        "--num-workers",
+        "1",
+        "--threadless",
+        "--hostname",
+        "0.0.0.0",
+        "--port",
+        str(port),
+        "--cert-file",
+        trustme_certs[0],
+        "--key-file",
+        trustme_certs[1],
```

Comment:
It seems like you could've just used
```suggestion
        tls_certificate_pem_path,
```